proposals/array_modify.json

```diff
@@ -29,15 +29,15 @@
         },
         {
             "name": "index",
-            "description": "The index of the element to insert the value(s) before. If the index is greater than the number of elements, the process throws an `ArrayElementNotAvailable` exception.\n\nTo insert after the last element, there are two options:\n\n1. Use the simpler processes ``array_append()`` to append a single value or ``array_concat()`` to append multiple values.\n2. Specify the number of elements in the array. You can retrieve the number of elements with the process ``count()``, having the parameter `condition` set to `true`.",
+            "description": "The index in the `data` array of the element to insert the value(s) before. If the index is greater than the number of elements in the `data` array, the process throws an `ArrayElementNotAvailable` exception.\n\nTo insert after the last element, there are two options:\n\n1. Use the simpler processes ``array_append()`` to append a single value or ``array_concat()`` to append multiple values.\n2. Specify the number of elements in the array. You can retrieve the number of elements with the process ``count()``, having the parameter `condition` set to `true`.",
             "schema": {
                 "type": "integer",
                 "minimum": 0
             }
         },
         {
             "name": "length",
-            "description": "The number of elements to replace. This parameter has no effect in case the given `index` does not exist in the array given.",
+            "description": "The number of elements in the `data` array to remove (or replace) at the given index. If there are not enough elements to remove, the process simply removes all elements after the given index.",
             "optional": true,
             "default": 1,
```

Comment:
As mentioned above (but probably lost in the discussion), I think a default `length=1`  is unnecessarily opinionated.

I would propose to use 0 by default, or not having a default at all

Reply:
I had a look but forgot to reply, sorry. I couldn't find a less opinionated alternative yet assuming that the main usecases are update and remove with having array_append and array_concat. I don't feel like there are a lot of common use cases for inserting in-between elements, but length = 0 would mostly cater for them. Length = null (i.e. length of values) would be okay, I think, but seems to make descriptions more complex. So if replacing length = 1 with length = null seems important enough for a change, then that would be my proposal.

Reply:
I don't think that the existence of `array_append` and `array_concat` is a reason to arbitrary pick default `length=1`. 

One could argue that default `length=0` is also opinionated, but at least it is the least destructive option (any length>0 removes data)

Another defense for in favor for the `length=0`  default is that the most obvious default array  is an empty array (length 0).

Reply:
Opened a new issue: #312